src/containers/views/DataFileEdit.js

```diff
@@ -51,51 +74,126 @@ export class DataFileEdit extends Component {
   }
 
   toggleView() {
-    this.setState({ guiView: !this.state.guiView });
+    const { datafile } = this.props;
+    this.setState({
+      guiView: !this.state.guiView,
+      guiPath: datafile.slug,
+      extn: datafile.ext
+    });
+  }
+
+  handleChange(e) {
+    const { onDataFileChanged } = this.props;
+    let obj = {};
+    const key = e.target.id;
+    const value = e.target.value;
+    obj[key] = value;
+
+    this.setState(obj);
+    onDataFileChanged();
   }
 
   handleClickSave(e) {
     const { datafile, datafileChanged, fieldChanged, putDataFile, params } = this.props;
+    const { path, relative_path } = datafile;
+    const data_dir = path.replace(relative_path, "");
```

Comment:
the convention I saw used in all `container/views/*` is that single quotes are used at the top when importing components, and double quotes within the current component class body.
So I think its best to handle all those files in a separate dedicated PR..